### ISSUE TITLE
Increasing gap with SciPy result

### ISSUE THREAD

**Issue by YifeiSun01** [tum-pbs/PhiFlow#196]
I tested the demo code on https://tum-pbs.github.io/PhiFlow/examples/grids/[Burgers](https://tum-pbs.github.io/PhiFlow/examples/grids/Burgers.html).html using both SciPy solve_ivp and  PhiFlow burgers. I noticed that there is a quite large gap between their results. Is this normal? I guess maybe they are implemented differently to solve the PDE, so their accumulated errors are different. Or maybe I am not using the code correctly. Thanks for any hint or advice!

This is the code I use:

from phi.jax.flow import *
from scipy.ndimage import gaussian_filter
import numpy as np
from scipy.integrate import solve_ivp
import matplotlib.pyplot as plt
from matplotlib.animation import FuncAnimation

class GRFGenerator:
    @staticmethod
    def generate_grf(shape, kernel='gaussian', kernel_params=None, bc='periodic', seed=None, zero_mean=False):
        if seed is not None:
            key = jax.random.PRNGKey(seed) if seed is not None else jax.random.PRNGKey(0)
        white_noise = jax.random.normal(key, shape=shape)
        
        if kernel == 'gaussian':
            cl = kernel_params.get('correlation_length', 1.0)*shape[0]
            if bc == 'periodic':
                mode = 'wrap'
            else:
                mode = 'nearest'
            grf = gaussian_filter(white_noise, sigma=cl, mode=mode)
        else:
            raise ValueError(f"Unknown kernel: {kernel}")
        
        grf = (grf - grf.min()) / (grf.max() - grf.min())
        if zero_mean == True:
            grf -= grf.mean()
        return grf

class SciPyBurgersSolver1D:
    def __init__(self, nx, nu, bc='periodic', num_int_method="RK45",conservative=False, xlim=(-1,1)):
        self.nx = nx
        self.x = np.linspace(xlim[0], xlim[1], nx)
        self.dx = (xlim[1]-xlim[0])/nx
        self.nu = nu
        self.bc = bc
        self.num_int_method = num_int_method
        self.conservative = conservative
    
    def compute_derivatives(self, u):
        d2u_dx2 = (np.roll(u, -1) - 2*u + np.roll(u, 1)) / self.dx**2
        if self.conservative == True:
            u_squared = u**2 / 2
            convective_term = (np.roll(u_squared, -1) - np.roll(u_squared, 1)) / (2 * self.dx)
            return convective_term, d2u_dx2
        else:
            du_dx = (np.roll(u, -1) - np.roll(u, 1)) / (2 * self.dx)
            return du_dx, d2u_dx2
    
    def dudt(self, t, u):
        if self.conservative == True:
            convective_term, d2u_dx2 = self.compute_derivatives(u)
            return -convective_term + self.nu * d2u_dx2
        else:
            du_dx, d2u_dx2 = self.compute_derivatives(u)
            return -u * du_dx + self.nu * d2u_dx2
    
    def solve(self, u0, t_final, t_eval, max_step):
        current_u = u0.copy()
        sol = solve_ivp(
            self.dudt, 
            [0, t_final], 
            current_u,
            t_eval=t_eval,
            method=self.num_int_method, 
            max_step=max_step,
            dense_output=True,
            rtol=1e-6, 
            atol=1e-8
        )
        return sol.y.T

grf_params = {
    'dim': 1,
    'nx': 2**11, 
    'kernel': 'gaussian',
    'kernel_params': {'correlation_length': 0.07}, 
    'bc': 'periodic', 
    'seed': 45,  
    'zero_mean': False 
}
grf = GRFGenerator.generate_grf(
    shape=(int(grf_params["nx"]),),  # Ensure tuple of integers
    kernel=grf_params['kernel'],
    kernel_params=grf_params['kernel_params'],
    bc=grf_params['bc'],
    seed=int(grf_params['seed']),
    zero_mean=grf_params['zero_mean']
) 

@jit_compile
def step(v, nu, dt=.5):
    v = diffuse.implicit(v, nu, dt=dt)
    v = advect.semi_lagrangian(v, v, dt=dt)
    return v

dt = 0.5
N = 2048
grf = grf[::int(len(grf)/N)]
nu = 0.1
xlim = (0,64)
step_num = 100

v0 = CenteredGrid(Noise(smoothness=2), PERIODIC, x=N, bounds=Box(x=xlim))
# v0 = CenteredGrid(math.tensor(grf, spatial('x')), PERIODIC, x=N, bounds=Box(x=xlim))

def step_fn(v):
    return step(v, nu=nu, dt=dt)

v_trj = iterate(step_fn, batch(time=step_num), v0)
solutions_1 = v_trj.values.numpy('x,vector').transpose(2,0,1).squeeze()

scipysolver = SciPyBurgersSolver1D(nx=N, nu=nu, num_int_method="RK45", conservative=False, xlim=xlim)
time_list = [i*dt for i in range(step_num+1)]
solutions_2 = scipysolver.solve(v0.values.numpy(), t_final=time_list[-1], t_eval=time_list, max_step=dt)

fig, ax = plt.subplots(figsize=(10, 6))
ax.set_xlim(xlim[0], xlim[1])
ax.set_ylim(min(np.min(solutions_1), np.min(solutions_2)),
            max(np.max(solutions_1), np.max(solutions_2)))
ax.set_xlabel('x')
ax.set_ylabel('v')
ax.grid(True)

line1, = ax.plot([], [], 'b', label='PhiFlow', lw=1)
line2, = ax.plot([], [], 'r', label='SciPy', lw=1)
time_text = ax.text(0.02, 0.95, '', transform=ax.transAxes)
ax.legend()
ax.set_title(f"N={N}")

x = np.linspace(xlim[0], xlim[1], N)

def init():
    line1.set_data([], [])
    line2.set_data([], [])
    time_text.set_text('')
    return line1, line2, time_text

def update(frame):
    line1.set_data(x, solutions_1[frame])
    line2.set_data(x, solutions_2[frame])
    time_text.set_text(f'Time = {time_list[frame]:.3f}')
    return line1, line2, time_text

ani = FuncAnimation(fig, update, frames=len(time_list),
                    init_func=init, blit=True, interval=100)

ani.save('burgers_comparison.gif', writer='pillow', fps=10, dpi=100)
plt.close()

![Image](https://github.com/user-attachments/assets/29e54049-8639-461a-adbd-b9319910f0dd)

**Comment by YifeiSun01** [comment]
![Image](https://github.com/user-attachments/assets/b0aeb663-3ed5-45b7-9a5d-ef8d1c0ab794)  This is the result I get.
